### DISCUSSION TITLE
Tasmota with client MCU

### DISCUSSION THREAD

**Discussion by Thom4fun** [arendst/Tasmota#17517] [Ideas]
I am using this great project to connect diffrent kind of MCU to the MQTT broker, which run like a charm.
Now, I am looking for the possiblity to request from the client CPU, the time from the tasmota device. 
Any Idea how to solve that?

Many thanks
Thomas

**Comment by joba-1** [comment]
not much detail, so general answer: use serial

**Reply by barbudor** [reply]
Are you using TasmotaClient?
How are you handling the discussion between the MCU and the ESP?
Solution will highly depends on that

**Reply by Thom4fun** [reply]
Thank you for your answer.
Yes I am using TasmotaClient. Since now, I can handle the comunication just to send the sensor data via MQTT to the broker.
Now I have the need to know the time for some calculations, but find no point how do I can get the time from the Tasmota device.

There is ClientSend which run well from the console. But Nothing to find hwo to pass a command, like time, and where to place.
As far as I can see in the source (where do I have just a minumum of experiance on C++) , as well with help from google, I cant find a solution. Nice would be to use ExecuteComand and get back the output of time() or to have an command to querry the Tasmota device.

AS well, when you say, use serial, I have no clue where to start on Tasmota side. Avery hint is very welcome....

Regards
Thomas

**Reply by sfromis** [reply]
I'd expect it to be possible to make a rule reacting to data received from the client, and then do a ClientSend to send the time, depending on what format is workable:
`%time%` for minutes since midnight
`%timestamp%` is formatted date/time
`%utctime%` is the Unix epoch

**Reply by Thom4fun** [reply]
I see what you are talking about. Let me have a try and I will report back.
Thank you for your suggestion.
